app/src/main/kotlin/com/wire/android/ui/home/conversations/MessageComposerViewModel.kt

```diff
@@ -105,12 +107,19 @@ class MessageComposerViewModel @Inject constructor(
     private val getAssetSizeLimit: GetAssetSizeLimitUseCase,
     private val sendKnockUseCase: SendKnockUseCase,
     private val pingRinger: PingRinger,
-    private val imageUtil: ImageUtil
+    private val imageUtil: ImageUtil,
+    private val fileManger: FileManager
 ) : SavedStateViewModel(savedStateHandle) {
 
     var conversationViewState by mutableStateOf(ConversationViewState())
         private set
 
+    var tempWritableVideoUri: Uri? = null
```

Comment:
Why are we declaring this uri as nullable? the `fileManager.getTempWritableVideoUri` is not a nullable one, so I wouldn't expect this one to be.

Reply:
We need to assign a default value before calling the suspend function `getTempWritableVideoUri`

Reply:
Wouldn't a lazy intialisation also work here? Otherwise it's also fine. I just don't want to make all the layers receive a nullable variable if it can be avoided.

Reply:
we can't call a suspend function from a lazy var and for the lateinit it will not work because that var needs to be initialised, otherwise it will crash